### ISSUE TITLE
Dialog: Cannot focus back to the button opening the dialog when the dialog is closed

### ISSUE THREAD

**Issue by cynthiahhc** [eBay/ebayui-core#1341]
The issue happens in Safari and Firefox (using OX), and it's in both 6.x.x and 5.x.x versions.

Usually, we open the dialog by clicking a button, but in Safari and Firefox, clicking the button will not focus on the button. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#clicking_and_focus

When `ebay-dialog-base` tries to cache the previously focused element before it opens by using `document.activeElement`, it caches the incorrect element. The button is never focus and when the dialog close, it will restore the focus to some other element, causing the page to scroll/bounce.

https://github.com/eBay/ebayui-core/blob/5.7.0/src/components/components/ebay-dialog-base/component.js#L126

**Comment by ianmcburnie** [comment]
Thank you for reporting this @cynthiahhc

**Comment by agliga** [comment]
Released in `@ebay/ebayui-core@5.7.13` and `@ebay/ebayui-core@7.1.2`
